var/spack/repos/builtin/packages/micromamba/package.py

```diff
@@ -0,0 +1,153 @@
+        depends_on("openssl", type=("link", "run"))
+        depends_on("yaml-cpp", type=("link", "run"))
+        depends_on("libreproc+cxx+shared", type=("link", "run"))
+        depends_on("tl-expected@b74fecd", type=("link", "run"))
+        depends_on("fmt", type=("link", "run"))
+        depends_on("spdlog", type=("link", "run"))
+        # See linkage=shared for usage location
+        depends_on("nlohmann-json", type="link")
+        depends_on("cpp-termcolor", type="link")
+        depends_on("cli11@2.2:", type="link")
+
+    with when("linkage=static"):
+        # When linkage is static, BUILD_STATIC=ON
+        # and then
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L523
+        # calls libmamba_create_target(libmamba-static STATIC SHARED libmamba)
+        # where the third argument, SHARED, is the deps_linkage
+        # as defined here,
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L256
+        # which would use dynamic linkage here,
+        # https://github.com/mamba-org/mamba/blob/micromamba-1.0.0/libmamba/CMakeLists.txt#L420
+        # See linkage=dynamic for what that entails.
+        depends_on("libsolv+conda", type=("link", "run"))
+        depends_on("curl libs=shared", type=("link", "run"))
+        depends_on("libarchive crypto=mbedtls xar=libxml2", type=("link", "run"))
+        depends_on("openssl", type=("link", "run"))
+        depends_on("yaml-cpp", type=("link", "run"))
+        depends_on("libreproc+cxx", type=("link", "run"))
+        depends_on("tl-expected@b74fecd", type=("link", "run"))
+        depends_on("fmt", type=("link", "run"))
+        depends_on("spdlog", type=("link", "run"))
+        depends_on("nlohmann-json", type="link")
+        depends_on("cpp-termcolor", type="link")
+        depends_on("cli11@2.2:", type="link")
+
+    patch("fix-threads.patch")
+
+    def cmake_args(self):
+        # See https://mamba.readthedocs.io/en/latest/developer_zone/build_locally.html#build-micromamba
+        if "linkage=dynamic" in self.spec:
+            linkage = "dynamic"
+        elif "linkage=static" in self.spec:
+            linkage = "static"
+        elif "linkage=full_static" in self.spec:
+            warnings.warn("Full static does not work yet!")
+            linkage = "full_static"
+        else:
+            raise ValueError(f"Unknown linkage type {self.spec}")
+        return [
+            self.define("BUILD_LIBMAMBA", True),
+            self.define("BUILD_MICROMAMBA", True),
+            self.define("BUILD_STATIC", linkage == "static"),
+            self.define("BUILD_STATIC_DEPS", linkage == "full_static"),
+            self.define("BUILD_SHARED", linkage == "dynamic"),
+            self.define("MICROMAMBA_LINKAGE", linkage.upper()),
+        ]
+
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def check_install(self):
```

Comment:
@tldahlgren what's the easiest way to support this test both at build time and post-install?